configuration/rules-blockly-run-and-process.md

```diff
@@ -0,0 +1,163 @@
+layout: documentation
+title: Rules Blockly - Run & Process
+---
+
+# Run & Process (Rules and Scripts)
+[return to Blockly Reference]({{base}}/configuration/rules-blockly.html)
+
+## Introduction
+
+This section contains several possibilities
+* Calling Rules or other scripts either located as files on the server or provided via the UI
+* Retrieving attributes provided by the rule context or from the own caller rule.
+* Transforming values via different conversion methods (map, regex, jsonpath)
+
+**A note about Rules and Scripts**
+
+A Script *is* a Rule too. It’s just a special type of rule that only has a single script action and a “Script” tag.
+
+{::options toc_levels="2..4"/}
+
+- TOC
+{:toc}
+
+{: #blockly-value-storage-overview}
+
+## Overview of the Run & Process blocks
+
+![run-and-process-blocks](images/blockly/blockly-run-and-process.png)
+
+## Run & Process Blocks
+
+### Call Script File
+
+![run-script-file](images/blockly/blockly-run-script.png)
+
+*Function:* Calls a script file with the given name which must be located in the *$OPENHAB_CONF/scripts/* folder.
+
+- No parameters can be provided
+
+### Run Rule or Script created in UI
+
+![run-rule-or-script](images/blockly/blockly-run-rule.png)
+
+*Function:* calls another *rule* or *script* that was created via the openHAB UI.
+
+The  *ruleUID* can be found in the list of rules or scripts in the second line of the list (like here *helloBlockly* or *helloWorld*)
+![rule-uuid](images/blockly/blockly-run-rule-uuid.png)
+
+Parameters are optional:
+
+* If not providing parameters leave the *with context* section as is.
+* If providing parameters you **must** replace the *create empty dictionary* block with [the *dictionary* block from the Lists section](https://community.openhab.org/t/blockly-reference/128785#lists-76), modifying the number of key/value pairs to suit.
+
+Notes:
+* parameters can be retrieved in the called script via the "get context attribute"-block (see below)
+* the called rule / script cannot return any value
+* it is not possible to access a particular function in another script or rule as scripts / rules have to be perceived as completely autonomous
+* the called rule / script is run synchronously
+
+### Retrieve context attribute from rule
+
+Function: Retrieve context attributes passed by a calling rule/script (see above)
+
+![](images/blockly/blockly-retrieve-context.png)
+
+This retrieves the value for the key "myKey1" that was passed by the calling rule.
+
+*Example*
+
+![retrieve-context-example](images/blockly/blockly-retrieve-context-example.png)
+
+### Retrieve rule context information
+
+![retrieve-context-info](images/blockly/blockly-retrieve-rule-context-info.png)
+
+*Function:* Retrieve event context related information of the rule
+
+A rule carries contextual information when triggered - this block can retrieve that information.
+
+* The rule trigger type defines which of the attributes are available to the rule - they will not all be available.
```

Comment:
Please use `-` for lists in all files.

Currently we have mixed usage of `-` and `*`.